ably/ably_test.go

```diff
@@ -65,3 +66,49 @@ func checkError(code int, err error) error {
 		return nil
 	}
 }
+
+func assertEquals(t *testing.T, expected interface{}, actual interface{}) {
+	if expected != actual {
+		t.Errorf("%v is not equal to %v", expected, actual)
+	}
+}
+
+func assertTrue(t *testing.T, value bool) {
+	if !value {
+		t.Errorf("%v is not true", value)
+	}
+}
+
+func assertFalse(t *testing.T, value bool) {
+	if value {
+		t.Errorf("%v is not false", value)
+	}
+}
+
+func assertNil(t *testing.T, object interface{}) {
+	value := reflect.ValueOf(object)
+	if !value.IsNil() {
+		t.Errorf("%v is not nil", object)
+	}
+}
+
+func assertNotNil(t *testing.T, object interface{}) {
+	value := reflect.ValueOf(object)
+	if value.IsNil() {
+		t.Errorf("%v is nil", object)
```

Comment:
I didn't intend to generalize it. Feel free to update it based on the requirement. In my case, it shows 
[ ] is nil. The same goes for the above example.

Reply:
> Also, an error message of `nil is nil` isn't very useful.

This is alleviated by calling `t.Helper()` from helper functions. Then the error feedback points to the line this function was called.

Not that I disagree that this function is not worth the trouble anyway. I think `assertEqual` is enough. Error checks can be just `if err !=nil { t.Fatal(err) }`, len checks can be just `assertEqual(y, len(x))`, etc.

Reply:
> In my case, it shows [ ] is nil.

I mean, that's even worse, no? That doesn't seem to convey anything useful to the caller, I'd much prefer to see the name of the variable rather than what it's value actually is (e.g. `foo is nil`, not `nil is nil` or `[] is nil`)

> Feel free to update it based on the requirement

I'll leave it with you to decide whether this needs to change, but whatever you go for I think should be documented in a comment on the function (i.e. if it stays as is, a comment that the error is non fatal and a further nil check should be made on the value if it's subsequently used).

Reply:
It's not being used, I will remove it for the time being